sdk/appconfiguration/azure-data-appconfiguration/samples/appconfig_basic_operation.cpp

```diff
@@ -74,6 +74,280 @@ static void RetreiveLabels(ConfigurationClient& configurationClient)
 #endif
 }
 
+// Retreive key values based on filters
+static void RetrieveConfigurationSettings(ConfigurationClient& configurationClient)
+{
+  // Current
+
+  {
+    GetKeyValuesOptions options;
+    options.Label = "production*";
+
+    for (GetKeyValuesPagedResponse keyValuesPage
+         = configurationClient.GetKeyValues("accept", options);
+         keyValuesPage.HasPage();
+         keyValuesPage.MoveToNextPage())
+    {
+      if (keyValuesPage.Items.HasValue())
+      {
+        std::vector<KeyValue> list = keyValuesPage.Items.Value();
+        std::cout << "KeyValues List Size: " << list.size() << std::endl;
+        for (KeyValue keyValue : list)
+        {
+          Azure::Nullable<std::string> valueOfKey = keyValue.Value;
+
+          if (valueOfKey.HasValue())
+          {
+            std::cout << keyValue.Key << " : " << valueOfKey.Value() << std::endl;
+          }
+          else
+          {
+            std::cout << "Value for: '" << keyValue.Key << "' does not exist." << std::endl;
+          }
+        }
+      }
+    }
+  }
+
+  // Expected
+
+#if 0
+  {
+    GetConfigurationSettingsOptions options;
+    options.Label = "production*";
+
+    for (GetConfigurationSettingsPagedResponse keyValuesPage
+         = configurationClient.GetConfigurationSettings(options);
+         keyValuesPage.HasPage();
+         keyValuesPage.MoveToNextPage())
+    {
+      if (keyValuesPage.Items.HasValue())
```

Comment:
It is beyond AppConfig, and this is not question only to you, but maybe we should have a design discussion with the architects, but I am wondering if we should make pageable items as non-nullable, at least, by default - I don't nullable here is the best user experience, and I don't expect nullable vs size == 0 to ever have meaningful distinction in the context of Azure SDK.
The spec says it is optional, and I get that, but if we don't take that literally, we could special-case that.
To be extra careful, we could make an option for that, which you could toggle to Off for the potential services where there is a meaningful distinction.

Reply:
That's an interesting question. I think I agree with you that Items should just be a `vector<T>` and not a `Nullable<vector<T>>`.

My one concern is that I suspect it is legal for a pageable to return 0 entries for a page even if there are more pages still to come (in other words, `HasPage` is true, but `Items` is empty. It is a coding mistake for a customer to treat `Items.empty()` as if it was the same as `HasPage()` returning false, but it would be an easy mistake for a developer to make. Making `Items` be nullable basically renders that mistake impossible at the cost of a poor developer experience.

Reply:
Hmm, but technically nothing prevents `pageable.Items.HasValue() == false` while `pageable.HasPage() == true`. And I also thought another possible mistake for the user to make - to assume that `Items.Value().size() > 0` if `Items.HasValue() == true`. So, it is complicated, for sure, but maybe we could make it a tiny bit better.
Whether the service returns `[]` or `null`/`undefined`, we can generate code to hide that from user, in all cases the result will be as if it was `[]`.

Reply:
To your point, @antkmsft, the existing hand-written SDKs don't use `Nullable<T>` for the collection: https://github.com/Azure/autorest.cpp/issues/475